### ISSUE TITLE
Show Information that something was written in a Thread in the Main Room

### ISSUE THREAD

**Issue by JokerGermany** [vector-im/element-meta#322]
#### What would you like to do?
When you get a notification for a thread there should be an information in the Main Room and when you click on it you get to the thread.

#### Why would you like to do it?
I think threads are confusing for some people.
You have a Notification for a room and when you go into the room there is nothing.
I don't think that all users will conclude that this is because of an event in a Thread.
You have to go to the threads and click onto the right thread to see the message.
I don't think this is user friendly.

#### How would you like to achieve it?
I don't know^^

**Comment by HarHarLinks** [comment]
I agree with your sentiment, I want to add

> You have to go to the threads

This is indicated by a grey or red (depending on ping) dot on the threads button top right. However I think this is hardly visible, and there should be a popup in the main room, similar to the "scroll to first unread" message or so.

> and click onto the right thread to see the message.

There is again a (hardly visible) dot beside unread threads. It would be better to highlight unread threads with a background color or so.


I think there was at least the idea that the thread with a new message opens automatically, but I don't know it that would actually be obvious enough when you're looking at the main TL.

**Comment by daniellekirkwood** [comment]
@janogarcia FYI 👀

**Comment by janogarcia** [comment]
@daniellekirkwood Yeah, the current implementation (MVP) is still missing a lot of the ideas we have for giving proper visibility to Thread updates.

**Comment by tobiasBora** [comment]
I also completely agree. To complete this idea, the message in the main room could also be like the message one gets when a new member joins a room, with a tiny text saying `A new message has been posted to thread XX`. The advantage of this is that it would not vanish (as opposed to unread messages), meaning that I can easily move to the appropriate thread later.

**Comment by janogarcia** [comment]
@tobiasBora Thanks for adding your feedback, Tobias. We actually do have some design explorations on improving the visibility of new activity in threads (updates in the left sidebar, thread activity digests in the timeline, etc.).

Still, we wouldn't be broadcasting every single update/new message in the timeline, as it would defeat the whole purpose of threads.

We may, however, expose it somehow in a less overwhelming way, as briefly mentioned in the note about the internal design explorations.

**Comment by tobiasBora** [comment]
Well, for you what is the purpose of threads? I can understand that some people may use threads to keep a cleaner timeline, especially in large IRC-like rooms, and that's the reason why I'm proposing a discrete greyed one-line message following the style of:

![image](https://user-images.githubusercontent.com/2164118/174314822-2e17f462-abbf-47a6-b7eb-ebdd972f502b.png)

but in my case I primarily see threads as a way to group answers together to ease the reading. IMHO, the timeline should be… well a timeline, i.e. an area that displays what happens when to allow people to quickly check what they missed in an eyesight (of course, I would prefer a small discrete note when a new message is written inside a thread in place of the whole message). Actually, I think I like the way Zulip handles threads, by moving down the whole thread (cut after a few messages) down whenever someone answers to the thread. This way, it's impossible to miss a message in a thread.

**Comment by janogarcia** [comment]
> Well, for you what is the purpose of threads?

Make multiple discussions more manageable over time. In rooms / for teams that choose to favor or combine async communication instead of relying on sync communication exclusively.

Zulip follows a threads-first paradigm, whereas Element implements a less opinionated approach to threads to support the existing user base better (work in parallel / side by side with sync chat rather than introducing extremely breaking changes on core interactions like replies).

We may, in the future, explore more structured/strict approaches like optional threads-only rooms or further explore the ideas I shared in my previous message.

**Comment by tobiasBora** [comment]
Sure, Zulip is definitely more opinionated (and it would be really cool to have an option to have thread-only rooms in matrix, especially if the current behavior stays possible), that's why I was thinking to use these mini notifications to provide something in between the current solution and the Zulip one (That said, one could also imagine a way to turn this notification on/off with an optional per-room granularity to make sure everyone can have the display they like).

Also, I'm not sure to see which note you refer to as `internal design`?

**Comment by janogarcia** [comment]
> Also, I'm not sure to see which note you refer to as `internal design`?

Sorry, I was referring to discussing it internally (company internal rooms) with the team that worked in the threads implementation. Just to give you some visual context (nothing here is final, and there's no guarantee we may end up implementing it at all).

**Exploration on threads events**

We could broadcast major events: A new thread is created on a relatively old message. The discussion has resumed in a thread that hadn't seen any new comments in a while.

![Thread events — Four new thread events happened, We only show the first three events in a bundle initially, the rest will be collapsed and accesible under a _Show x more_ toggle](https://user-images.githubusercontent.com/60790/174323705-e6547a7f-c3aa-4e3b-9d6e-5d232b6baaf1.png)

**Exploration on thread activity updates/navigation on the left sidebar**

https://github.com/vector-im/element-web/issues/19615#issuecomment-1026158294

It's not Zulip, but both explorations combined with thread-only rooms could get you to a somewhat closer experience.

**Comment by tobiasBora** [comment]
Thanks for sharing this! So the "Exploration on threads events" is basically what I was referring to, except that I was thinking to smaller notifications (notably when a discussion is just resumed) and that the "relatively old" is defined as 0 seconds :-P (I find it a bit weird to have a notion of "relatively old", as if I don't see a notification I will never know if it's because the threads were not old enough or just because there were no activity). Also, an in-between option would be to display this event only for the threads in which you personally participated, even if I'm more in favor of putting events for everything. But I guess one could also let the user configure it the way they want.
